ansible/README.md

```diff
@@ -0,0 +1,121 @@
+Deploying Openwhisk using Ansible
+=========
+
+
+### Getting started
+
+If you want to deploy Openwhisk locally using Ansible, you first need to install Ansible on your development environment:
+
+#### Ubuntu users
+```
+sudo apt-get install python-pip
+sudo pip install ansible==2.0.2.0
+```
+
+#### Vagrant users
+Nothing to be done, Ansible is already installed during vagrant provisioning.
+
+#### Mac users
+It is assumed that a VM has been provisioned using **docker-machine**.
+
+```
+brew install python
+pip install ansible==2.0.2.0
+
+cd ansible
+ansible-playbook setup.yml
+```
+
+After this there should be a `hosts` file in the `ansible/environments/mac` directory.
+
+To verify the hosts file you can do a quick ping to the docker machine:
+
+```
+cd ansible
+ansible all -i environments/mac -m ping
+```
+
+Should result in something like:
+
+```
+ansible | SUCCESS => {
+    "changed": false,
+    "ping": "pong"
+}
+192.168.99.100 | SUCCESS => {
+    "changed": false,
+    "ping": "pong"
+}
+```
+
+### Using Ansible
+**Caveat:** All ansible commands are meant to be executed from the `ansible` directory.
+This is important because that's where `ansible.cfg` is located which contains generic settings that are needed for the remaining steps.
+
+In all instructions, replace `<environment>` with your target environment. e.g. `mac`if you want to deploy using a local mac setup.
+
+In all instructions, replace `<openwhisk_home>` with the base directory of your Openwhisk source tree. e.g. `openwhisk`
+
+
+#### Install Prerequisites
+This needs to be done only once per environment. It will install necessary prerequisites on all target hosts in the environment.
+
+```
+ansible-playbook -i environments/<environment> prereq.yml
```

Comment:
actually, it works as designed. It has to check if `pip` is installed in the target at first by invoking the `pip` command (valuate return code), if it's not, it will install pip.

However, ansible treats every **non zero** return code as a failure and stops the process. so, we have to ignore it.

Another idea, we can install `pip` in the target machine by forcing, but IMHO it's not good to overwrite pip of user installation.